src/external_app/external_app_entrypoint.ts

```diff
@@ -64,6 +65,14 @@ const handleExternalMessage = (
       success: true,
       result: null,
     });
+  } else if (msg.command === "navigate") {
+    navigate(msg.payload.path, msg.payload.options);
+    bus.fireMessage({
+      id: msg.id,
+      type: "result",
+      success: true,
+      result: null,
+    });
```

Comment:
Navigate could return false if there is an open dialog that takes too long to close. 
Don't know what you do with the result but you should send success with the return value of navigate.